pkg/dockerfile/fast_generator.go

```diff
@@ -0,0 +1,271 @@
+	}
+	path := filepath.Join(tmpDir, filename)
+	if err := os.MkdirAll(filepath.Dir(path), 0o755); err != nil {
+		return "", fmt.Errorf("Failed to write %s: %w", filename, err)
+	}
+	if err := os.WriteFile(path, data, 0o644); err != nil {
+		return "", fmt.Errorf("Failed to write %s: %w", filename, err)
+	}
+	return path, nil
+}
+
+func (g *FastGenerator) generateMonobase(lines []string, tmpDir string) ([]string, error) {
+	lines = append(lines, []string{
+		"# syntax=docker/dockerfile:1-labs",
+		"FROM r8.im/monobase:latest",
+	}...)
+
+	cogPath, err := g.copyCog(tmpDir)
+	if err != nil {
+		return nil, err
+	}
+
+	lines = append(lines, []string{
+		"ENV R8_COG_VERSION=\"file:///buildtmp/" + filepath.Base(cogPath) + "\"",
+	}...)
+
+	if g.Config.Build.GPU {
+		cudaVersion := g.Config.Build.CUDA
+		cudnnVersion := g.Config.Build.CuDNN
+		lines = append(lines, []string{
+			"ENV R8_CUDA_VERSION=" + cudaVersion,
+			"ENV R8_CUDNN_VERSION=" + cudnnVersion,
+			"ENV R8_CUDA_PREFIX=https://monobase.replicate.delivery/cuda",
+			"ENV R8_CUDNN_PREFIX=https://monobase.replicate.delivery/cudnn",
+		}...)
+	}
+
+	lines = append(lines, []string{
+		"ENV R8_PYTHON_VERSION=" + g.Config.Build.PythonVersion,
+	}...)
+
+	torchVersion, ok := g.Config.TorchVersion()
+	if ok {
+		lines = append(lines, []string{
+			"ENV R8_TORCH_VERSION=" + torchVersion,
+		}...)
+	}
+
+	buildTmpMount, err := g.buildTmpMount(tmpDir)
+	if err != nil {
+		return nil, err
+	}
+
+	return append(lines, []string{
+		"RUN " + strings.Join([]string{
+			buildTmpMount,
+			g.monobaseUsercacheMount(),
+			APT_CACHE_MOUNT,
+			UV_CACHE_MOUNT,
+		}, " ") + " UV_CACHE_DIR=\"" + UV_CACHE_DIR + "\" /opt/r8/monobase/run.sh monobase.build --mini --cache=" + MONOBASE_CACHE_PATH,
+	}...), nil
+}
+
+func (g *FastGenerator) copyWeights(lines []string, weights []Weight) ([]string, error) {
+	if len(weights) == 0 {
+		return lines, nil
+	}
+
+	for _, weight := range weights {
+		lines = append(lines, "COPY --link \""+weight.Path+"\" \""+filepath.Join(FUSE_RPC_WEIGHTS_PATH, weight.Digest)+"\"")
+	}
+
+	return lines, nil
+}
+
+func (g *FastGenerator) install(lines []string, weights []Weight, tmpDir string) ([]string, error) {
+	// Install apt packages
+	packages := g.Config.Build.SystemPackages
+	if len(packages) > 0 {
+		lines = append(lines, "RUN "+APT_CACHE_MOUNT+" apt-get update && apt-get install -qqy "+strings.Join(packages, " ")+" && rm -rf /var/lib/apt/lists/*")
+	}
+
+	// Install python packages
+	requirementsFile, err := g.pythonRequirements(tmpDir)
+	if err != nil {
+		return nil, err
+	}
+	buildTmpMount, err := g.buildTmpMount(tmpDir)
+	if err != nil {
+		return nil, err
+	}
+	if requirementsFile != "" {
+		lines = append(lines, "RUN "+strings.Join([]string{
+			buildTmpMount,
+			g.monobaseUsercacheMount(),
+			UV_CACHE_MOUNT,
+		}, " ")+" UV_CACHE_DIR=\""+UV_CACHE_DIR+"\" /opt/r8/monobase/run.sh monobase.build --requirements=/buildtmp/requirements.txt --cache="+MONOBASE_CACHE_PATH)
```

Comment:
Should be `monobase.user` once https://github.com/replicate/monobase/pull/63 is merged.

Should also add `UV_COMPILE_BYTECODE=0` for this one step, otherwise UV compiles byte code of the managed Python interpreter in `/srv/r8/monobase/uv/python`, polluting that directory.

Reply:
Resolved in [339abfd](https://github.com/replicate/cog/pull/2108/commits/339abfd2ace253151e2726b6fef0b47264d7e469)